hdfs-commons/src/main/java/org/apache/mesos/hdfs/file/FileUtils.java

```diff
@@ -26,14 +27,20 @@ public static void createDir(File dataDir) {
   /**
    * Delete a file or directory.
    */
-  public static boolean deleteFile(File fileToDelete) {
-    if (fileToDelete.isDirectory()) {
-      String[] entries = fileToDelete.list();
-      for (String entry : entries) {
-        File childFile = new File(fileToDelete.getPath(), entry);
-        deleteFile(childFile);
+  public static boolean deleteDirectory(File fileToDelete) {
+    boolean deleted = false;
+
+    try {
+      if (fileToDelete.isDirectory()) {
```

Comment:
my only challenge with that is there is nothing HDFS about it.  It is simple a utility class pulling out code that was common in the code base.   that said... I would be happy to do that if that is what you want to see.